Support brotli content-encoding

ato opened the pull request:
Adds support for `Content-Encoding: br` which is now broadly supported by browsers and turning up in WARCs created by tools like Webrecorder.

ldko commented:
Hey @ato, thanks for submitting this! I looked over it somewhat quickly, and I think it looks good, but I can look at it more later. Initial thoughts: perhaps update the Javadoc comment for `decodeResource` in TextReplayRenderer.java, update the [release notes](https://github.com/iipc/openwayback/blob/master/src/site/markdown/release_notes.md), and maybe add a test in `DecodingResourceTest` to hit the `default` case in the `switch` statement.

ato commented:
Thanks for the review @ldko. Updated to address your comments.

Adding the tests made me realise the existing fallback behaviour when `Content-Encoding: gzip` is present but the content is not actually gzipped was losing the first 2 bytes of the stream. So I fixed that too. I'm not sure that fallback behaviour is worthwhile (I wonder if browsers actually handle false content-encoding headers or not) but since it already exists I didn't want to break it.

Sorry for the force-push. I messed up resolving the release-notes conflict (Windows line endings gah!) and so reset and rebased to clean up.
